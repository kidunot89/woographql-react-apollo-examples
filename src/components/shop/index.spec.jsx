import { render, waitFor } from "@testing-library/react";
import { MemoryRouter } from "react-router-dom";

import ApolloMockProvider from "../../testing/apollo-mock-provider";
import ProductsList from ".";

describe("ProductsList component", () => {
  const { baseElement, queryByText } = render(
    <ApolloMockProvider>
      <MemoryRouter>
        <ProductsList first={5} />
      </MemoryRouter>
    </ApolloMockProvider>
  );

  it("renders successfully", async() => {
    await waitFor(() => expect(queryByText(/Fetching products/)).toBeTruthy());
    await waitFor(() => expect(queryByText(/Fetching products/)).toBeNull());
    expect(baseElement).toMatchSnapshot();

  });
});
