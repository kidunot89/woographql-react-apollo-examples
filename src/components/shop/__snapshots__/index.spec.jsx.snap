// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ProductsList component renders successfully 1`] = `
<body>
  <div>
    <div
      class="container mx-auto relative py-3 px-4 max-w-full grid grid-cols-1 gap-y-8 md:grid-cols-3 lg:max-w-4xl lg:grid-cols-4 lg:gap-4"
    >
      <a
        class="mx-auto border border-shark rounded flex flex-col justify-center no-underline group"
        href="/"
      >
        <img
          alt=""
          class="max-w-full w-44"
          src="https://example.com/content/uploads/2021/04/tea-tumbler.png"
        />
        <div
          class="inline-flex flex-col justify-center items-center shrink w-44 h-44"
        >
          <div
            class="mt-0.5 mx-0.5 mb-0 text-base text-center transition-colors duration-500 group-hover:text-persian-green"
          >
            Tea Tumbler
            <br />
            <small
              class="text-lynch"
            >
              Encourages hydration
              <br />
              
Portable
            </small>
            

          </div>
        </div>
      </a>
      <a
        class="mx-auto border border-shark rounded flex flex-col justify-center no-underline group"
        href="/"
      >
        <img
          alt=""
          class="max-w-full w-44"
          src="https://example.com/content/uploads/2021/04/9781849754323.png"
        />
        <div
          class="inline-flex flex-col justify-center items-center shrink w-44 h-44"
        >
          <div
            class="mt-0.5 mx-0.5 mb-0 text-base text-center transition-colors duration-500 group-hover:text-persian-green"
          >
            Vegetarian Tagines & Couscous
            <br />
            <small
              class="text-lynch"
            >
              Inspires cooking
              <br />
              
Supports wellness
            </small>
            

          </div>
        </div>
      </a>
      <a
        class="mx-auto border border-shark rounded flex flex-col justify-center no-underline group"
        href="/"
      >
        <img
          alt=""
          class="max-w-full w-44"
          src="https://example.com/content/uploads/2021/04/GIVE-HATS-MEN-square.png"
        />
        <div
          class="inline-flex flex-col justify-center items-center shrink w-44 h-44"
        >
          <div
            class="mt-0.5 mx-0.5 mb-0 text-base text-center transition-colors duration-500 group-hover:text-persian-green"
          >
            Gift Of Hats (Men)
            <br />
            <small
              class="text-lynch"
            >
              Covers the whole head
              <br />
              
Protects against exposure
            </small>
            

          </div>
        </div>
      </a>
      <a
        class="mx-auto border border-shark rounded flex flex-col justify-center no-underline group"
        href="/"
      >
        <img
          alt=""
          class="max-w-full w-44"
          src="https://example.com/content/uploads/2021/04/Product7_Transparent.png"
        />
        <div
          class="inline-flex flex-col justify-center items-center shrink w-44 h-44"
        >
          <div
            class="mt-0.5 mx-0.5 mb-0 text-base text-center transition-colors duration-500 group-hover:text-persian-green"
          >
            Organic Lavender Hand Sanitizer Spray
            <br />
            <small
              class="text-lynch"
            >
              Helps reduce bacteria
              <br />
              
Calming aroma
            </small>
            

          </div>
        </div>
      </a>
      <a
        class="mx-auto border border-shark rounded flex flex-col justify-center no-underline group"
        href="/"
      >
        <img
          alt=""
          class="max-w-full w-44"
          src="https://example.com/content/uploads/2021/04/essential-oils-bundle.png"
        />
        <div
          class="inline-flex flex-col justify-center items-center shrink w-44 h-44"
        >
          <div
            class="mt-0.5 mx-0.5 mb-0 text-base text-center transition-colors duration-500 group-hover:text-persian-green"
          >
            Aromatherapy Essential Oil Bundle
            <br />
            <small
              class="text-lynch"
            >
              Creates a relaxing environment
              <br />
              
Enhances mood
            </small>
            

          </div>
        </div>
      </a>
    </div>
    <button
      class="cta-button fixed mb-4 bottom-0 left-1/2 -translate-x-1/2 w-1/2"
    >
      Load More
    </button>
  </div>
</body>
`;
